qmetaobject/qmetaobject_rust.hpp

```diff
@@ -21,23 +21,106 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #include <QtCore/QEvent>
 #include <QtCore/QDebug>
 
+/// Raw representation of a pointer to member function as specified by Itanium ABI.
+/// This is somehow similar to `std::raw::TraitObject` in Rust, but for C++.
+///
+/// Note: it can not be used to represent a pointer to data member.
+///
+/// http://itanium-cxx-abi.github.io/cxx-abi/abi.html#member-pointers
+struct MemberFunctionPtr {
```

Comment:
Hmm... Interesting. I'll look into it. I thought all modern compilers follow the Itanium "standard".

Reply:
https://devblogs.microsoft.com/oldnewthing/20040209-00/?p=40713

So, basically a pointer to a member function boils down to just a pointer to a function whenever it can do so. Which means, adjusting part is not used, and essentially can be assumed to be zero, AND it is always assumed to be zero by the current implementation of qmetaobject-rs. Which is why it was just `qintptr` before, and the second (adj) part was already initialized to zero because union initializes to the first element's initializer which was a pointer to a member function, but which could have been as large as a `sizeof(void *)` in which case would not even exist. Oof...